### ISSUE TITLE
Add option to keep all table descriptor leases active on all nodes

### ISSUE THREAD

**Issue by jseldess** [cockroachdb/docs#3400]
Background: https://airtable.com/tblD3oZPLJgGhCmch/viw8IfZMg9Up8SP33/recmeOiJ81qstDJ6A
Eng partners: @vivekmenezes, @dt

**Comment by vivekmenezes** [comment]
I'm creating a mechanism for this which is automatic so this will not need docs support. We might add some configuration options in for 2.2.

**Comment by jseldess** [comment]
OK, thanks, @vivekmenezes. I heard "option" as something users can control. But even if there's no user action involved, we'll probably want to revise/improve this part of our architecture docs: https://www.cockroachlabs.com/docs/v2.0/architecture/replication-layer.html#expiration-based-leases-meta-system-ranges.

**Comment by jseldess** [comment]
Actually, I'm not sure that part (or any part) of the architecture docs cover table descriptors...

**Comment by jseldess** [comment]
PR: https://github.com/cockroachdb/cockroach/pull/28725

From release notes:
> CockroachDB now periodically refreshes table leases to avoid initial latency on tables that have not been accessed recently. [#28725][#28725] {% comment %}doc{% endcomment %}

There's also a cluster setting for changing the refresh limit:

```
sql.tablecache.lease.refresh_limit                   | 50        | i            | maximum number of tables to periodically refresh leases for
```

@vivekmenezes, I assume we should not call out that setting in the docs? In that case, do you agree that we should add a sentence/paragraph about table leases in the architecture docs?

**Comment by vivekmenezes** [comment]
I don't think we want to document this because this is a temporary fix for a problem. In release 2.2 we plan to implement epoch based leases which will get rid of this entire refresh mechanism. IMHO it's best the user not know about table leases at all.

**Comment by vivekmenezes** [comment]
and we could mention them in the architecture docs for 2.2

**Comment by jseldess** [comment]
OK, makes sense to me. Moving to 2.2.
